### PR TITLE
Undo redo

### PR DESCRIPTION
**Pull request by shanexavier313**
Implement undo-redo by tracking state in `past` and `future` Editor class member arrays.

Logic is as follows:
- upon initializing, set `past` to the current state
- every time `componentDidUpdate()` is called check if the state is different and if so, push it to `past`
- `undo()` pops the current state out of `past` and places it in `future`. It then sets the state to the previous state before that using `updateStory()` + `forceUpdate()`.
- `redo()' pops the most recently 'undone' state (or 'future' state) off of future and sets it as state in a similar fashion.

undo_redo_demo

Notes:
- while functional, using `updateStory()` and `forceUpdate()` is kind of the hacky way to do this, resulting in multiple unnecessary calls to `componentDidUpdate()`. There's some weirdness about how state is being updated in the app so should investigate further to make the setting of state more effecient.

### CODE REVIEW

File: .tool-versions [hunk 1]
```diff
@@ -1 +1 @@
-ruby 2.5.1
+ruby 2.6.4
```
Comment (efatsi) [inline .tool-versions:1]: revert this as it was just for your local setup

Comment (efatsi) [reply .tool-versions:1]: probably worth getting 2.5.1 running locally

File: app/javascript/src/components/Editor.tsx [hunk 2]
```diff
@@ -20,6 +20,9 @@ import { StateConsumer, ApplicationState } from '../Store'
 import { save } from '../persistance'
 import { clone } from '../clone'
 
+// import CustomDiagramModel from '../CustomDiagramModel'
+// import CustomNodeModel from '../CustomNodeModel'
+
```
Comment (efatsi) [inline app/javascript/src/components/Editor.tsx:25]: delete

File: Gemfile.lock [hunk 3]
```diff
@@ -268,7 +268,7 @@ DEPENDENCIES
   webpacker (~> 4.0.7)
 
 RUBY VERSION
-   ruby 2.5.1p57
+   ruby 2.6.4p104
```
Comment (efatsi) [inline Gemfile.lock:271]: missed a spot

**Comment by shanexavier313** [comment]
Yeah it was, undid the commit

**Pull request merged**